src/app/dashboard/(admin)/ongoing-campaigns/[campaign-id]/edit/page.tsx

```
@@ -87,26 +100,92 @@ export default function EditCampaignPage() {
+
+      setInitialData(formData);
+      setIsSaveModalOpen(false);
+      setShowSuccessToast(true);
+    } catch (err) {
+      const message = err instanceof Error ? err.message : "An unknown error occurred.";
+      console.error("Save error:", message);
+      setSaveErrorMessage(message);
+      setIsSaveModalOpen(false);
+      setShowErrorToast(true);
+    }
+  }, [formData, initialData, parsedCampaignId, campaignEditData, updateCampaignMutation, queryClient]);
```

ESLint

[error] Compilation Skipped: Existing memoization could not be preserved

React Compiler has skipped optimizing this component because the existing manual memoization could not be preserved. The inferred dependencies did not match the manually specified dependencies, which could cause the value to change more or less frequently than expected. The inferred dependency was `setMainImageMutation`, but the source dependencies were [formData, initialData, parsedCampaignId, campaignEditData, updateCampaignMutation, queryClient]. Inferred different dependency than source.

/home/jailuser/git/src/app/dashboard/(admin)/ongoing-campaigns/[campaign-id]/edit/page.tsx:107:41
  105 |   }, []);
  106 |
> 107 |   const handleConfirmSave = useCallback(async () => {
      |                                         ^^^^^^^^^^^^^
> 108 |     if (!parsedCampaignId) return;
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 109 |
      …
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 181 |     }
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 182 |   }, [formData, initialData, parsedCampaignId, campaignEditData, updateCampaignMutation, queryClient]);
      | ^^^^ Could not preserve existing manual memoization
  183 |
  184 |   const navigateToCampaignPage = useCallback(() => {
  185 |     if (!parsedCampaignId) {

(react-hooks/preserve-manual-memoization)